src/App.tsx

```diff
@@ -15,47 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+  RESET,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortBy] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  let isResetButton = false;
+
+  let visibleGoods = [...goodsFromServer];
+
+  switch (sortType) {
+    case SortType.NONE:
+      isResetButton = false;
+      break;
+
+    case SortType.ALPHABET:
+      visibleGoods = visibleGoods.sort(
+        (a, b) => (a.localeCompare(b.toString())
+        ),
+      );
+      isResetButton = true;
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods = visibleGoods.sort(
+        (a, b) => (a.length - b.length),
+      );
+      isResetButton = true;
+      break;
+
+    case SortType.RESET:
+      visibleGoods = [...goodsFromServer];
+      isResetButton = false;
+      break;
```

Comment:
```suggestion
    case SortType.RESET:
      visibleGoods = [...goodsFromServer];
      isResetButton = false;
      break;
```
no need to make SortType Reset and  variable isResetButton to change after each change SortType;

just make function for reset button which set SortType.None and isReversed in false.
and count isResetButton something like that one time
const isResetButton = isReversed || sortType !== SortType.NONE

Reply:
const isResetButton = isReversed || sortType !== SortType.NONE 
 if SortType === Reset => isResetButton should be invisible, so I made another function : 
 const isResetButton = isReversed
  || sortType === SortType.ALPHABET
  || sortType === SortType.LENGTH;